typescript/sdk/src/ism/utils.ts

```diff
@@ -38,6 +39,46 @@ import {
 
 const logger = rootLogger.child({ module: 'IsmUtils' });
 
+// Determines the domains to enroll and unenroll to update the current ISM config
+// to match the target ISM config.
+export function calculateDomainRoutingDelta(
```

Comment:
if you use `Sets` you can leverage these
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/465112db6fddb3b598d6da39c13491ff1bc7e700/typescript/utils/src/sets.ts to greatly simplify this functino

Reply:
I'm not totally sure - we'd have to do a deep comparison of configs per domain as well and Sets do a dumb comparison instead of the deep equals we'd want

times like this i wish everyone used C# so we could make everything structs and implement comparators and reuse libs for free :P